web/src/main/java/org/springframework/security/web/authentication/UsernamePasswordAuthenticationFilter.java

```diff
@@ -74,10 +74,10 @@ public Authentication attemptAuthentication(HttpServletRequest request, HttpServ
 		if (this.postOnly && !request.getMethod().equals("POST")) {
 			throw new AuthenticationServiceException("Authentication method not supported: " + request.getMethod());
 		}
+
 		String username = obtainUsername(request);
-		username = (username != null) ? username.trim() : "";
```

Comment:
For backward compatibility, this line needs to stay where it is. If someone is overriding obtainUsername today, then this change would mean when they upgrade, the trimming would stop.

If you need it to be switched off perhaps we could add a property to toggle the trimming behavior.

Reply:
The intention behind my pull request was to ensure that methods starting with 'obtain' always return a valid string (or an empty string) rather than potentially returning null. I believed this approach would lead to more robust and predictable behavior. However, as you've pointed out, I now realize that I may not have adequately considered backward compatibility issues in my proposed changes.